tests/test_shared.py

```diff
@@ -0,0 +1,37 @@
+"""Unittests for internal shared code in test/shared.py"""
```

Comment:
could these new tests go in `tests/test_other.py`? `tools/shared.py` is imported there already, and that way these tests would get run automatically.

Reply:
My thinking was that most of the current tests are system tests, that test from the outside, where as these are actual unittests testing from within.  But maybe I'm wrong and there are already some unittests in test_other.

Reply:
Almost entirely, yeah, but we do have some unit or unit-ey tests there already, like `test_cmake_bitcode_static_libraries`. `shared` etc. are imported and used there in various ways.

I'd prefer to add new tests there, as otherwise adding another suite means a bunch of docs to update, bot changes, etc. But on the other hand if we think we want to add a lot more unit tests, maybe it's worth it, so I don't feel strongly either way.